daemon/dns/dns_resolvd.go

```diff
@@ -0,0 +1,177 @@
+package dns
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+	"strings"
+
+	"github.com/NordSecurity/nordvpn-linux/internal"
+)
+
+// Daemons and Services
+const (
+	// serviceSystemdResolved defines resolved service managed by systemd
+	serviceSystemdResolved = "systemd-resolved"
+)
+
+// Executables
+const (
+	// execBusctl defines busctl executable
+	execBusctl = "busctl"
+)
+
+// Systemd-resolved DBUS API based DNS handling method
+type Resolved struct{}
+
+func (m *Resolved) Set(iface string, nameservers []string) error {
+	return setDNSWithSystemdResolve(iface, nameservers)
+}
+
+func (m *Resolved) Unset(iface string) error {
+	return unsetDNSWithSystemdResolve(iface)
+}
+
+func (m *Resolved) IsAvailable() bool {
+	return internal.IsServiceActive(serviceSystemdResolved)
+}
+
+func (m *Resolved) Name() string {
+	return "resolved"
+}
+
+// setDNSWithSystemdResolve uses systemd-resolve dbus API to manage DNS
+// https://www.freedesktop.org/wiki/Software/systemd/resolved/
+func setDNSWithSystemdResolve(ifname string, addresses []string) error {
```

Comment:
The [docs](https://github.com/poettering/systemd/commit/8cedb0aef94da880e61b4c8cfeb7f450f8760ec6) suggest to use:
```
SetLinkDomains()
SetLinkDefaultRoute()
```
for the default route, might be optional but:
```
If managing a *privacy* VPN, you include `~.` in the routing domains,
 the value for `default-route` is actually irrelevant, but I'd set it to true. 
No need to configure any reverse lookup routing domains for it.
```

Reply:
Done